src/components/knowledge-network/KnowledgeNetworkPage.tsx

```
@@ -0,0 +1,386 @@
+import { useAuth } from '../../contexts/AuthContext';
+import useKnowledgeGraph from '../../hooks/useKnowledgeGraph';
+import { Loader2, AlertTriangle, X, Calendar, User, Tag } from 'lucide-react';
+import { supabase } from '../../lib/supabase';
+
+const ForceGraph2D = React.lazy(() => import('react-force-graph-2d'));
+
+export const KnowledgeNetworkPage: React.FC = () => {
+  const { user } = useAuth();
+  const { data, isLoading, error } = useKnowledgeGraph();
+  const [isMounted, setIsMounted] = useState(false);
+  const [selectedNode, setSelectedNode] = useState<any>(null);
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)